### PR TITLE
PS-8556 Jenkins MTR run should also allow to run tests requiring ps-p…

### PR DESCRIPTION
**Pull request by surbhat1595**
…rotocol

### CODE REVIEW

File: jenkins/param-parallel-mtr.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 - job:
-    name: percona-server-8.0-param-parallel-mtr
+    name: percona-server-8.0-param-parallel-mtr-ps8556
```
Comment (inikep) [inline jenkins/param-parallel-mtr.yml]: I think we don't want to change job name. In all places it should stay `percona-server-8.0-param-parallel`.

File: jenkins/param-parallel-mtr.yml [hunk 2]
```diff
@@ -211,7 +217,7 @@
           - debian:bookworm
     builders:
     - trigger-builds:
-      - project: percona-server-8.0-pipeline-parallel-mtr
+      - project: percona-server-8.0-pipeline-parallel-mtr-ps8556
```
Comment (inikep) [inline jenkins/param-parallel-mtr.yml]: In all 4 places it should stay `percona-server-8.0-pipeline-parallel`.

File: local/test-binary-parallel-mtr [hunk 3]
```diff
@@ -333,6 +333,69 @@ if [[ "${CI_FS_MTR}" = 'yes' ]]; then
...
+    start=`date +%s`
+
+    MTR_ARGS_PS=${MTR_ARGS}
+
+    if [[ $UNIT_TESTS == "1" ]]; then
+        MTR_ARGS_PS+=" --unit-tests-report"
+        UNIT_TESTS=0
+    fi
+    # no --mem allowed if vardir specified
+    MTR_ARGS_PS=${MTR_ARGS_PS//"--mem"/""}
```
Comment (inikep) [inline local/test-binary-parallel-mtr]: Probably not required.

File: local/test-binary-parallel-mtr [hunk 4]
```diff
@@ -333,6 +333,69 @@ if [[ "${CI_FS_MTR}" = 'yes' ]]; then
...
+
+    if [[ $BIG_TEST == "1" ]]; then
+        MTR_ARGS_PS+=" --big-test"
+    fi
+
+    # Collect all PS_PROTOCOL tests
+    PS_TESTS=$(grep --exclude="*.inc" --exclude="*.log" -rl . -e include/have_ps_protocol.inc | awk -F '/' '{print $(NF-2)"."$NF}' | sed 's/\.test//g' | sed 's/^\./main/g')
+    # Requested in PS-7602
```
Comment (inikep) [inline local/test-binary-parallel-mtr]: Please remove this and next comment lines.

**Review by inikep — CHANGES_REQUESTED** [review]

File: local/test-binary-parallel-mtr [hunk 5]
```diff
@@ -333,6 +333,69 @@ if [[ "${CI_FS_MTR}" = 'yes' ]]; then
...
+if [[ "${WITH_PS_PROTOCOL}" = 'yes' ]]; then
+    echo "Running PS PROTOCOL tests"
+    start=`date +%s`
+
+    MTR_ARGS_PS=${MTR_ARGS}
+
+    if [[ $UNIT_TESTS == "1" ]]; then
+        MTR_ARGS_PS+=" --unit-tests-report"
+        UNIT_TESTS=0
+    fi
```
Comment (inikep) [inline local/test-binary-parallel-mtr]: This should be removed after https://github.com/Percona-Lab/ps-build/pull/401 was merged

**Review by inikep — CHANGES_REQUESTED** [review]
You didn't address my comments regarding to removing `-ps8556` from the pipeline/param name.
The rest looks good.

**Comment by inikep** [comment]
Probably it will be good to rebase your branch to current `Percona-Lab:8.0` to solve conflicts.

**Review by inikep — APPROVED** [review]
LGTM

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**